rdeditor/rdEditor.py

```diff
@@ -209,242 +215,387 @@ def closeEvent(self, event):
         event.ignore()
 
     def exitFile(self):
-        response = self.msgApp("Confirmation","This will quit the application. Do you want to Continue?")
+        response = self.msgApp(
+            "Confirmation", "This will quit the application. Do you want to Continue?"
+        )
         if response == "Y":
             self.ptable.close()
-            exit(0) #TODO, how to exit qapplication from within class instance?
+            exit(0)  # TODO, how to exit qapplication from within class instance?
         else:
             self.editor.logger.debug("Abort closing")
 
-
     # Function to show Diaglog box with provided Title and Message
-    def msgApp(self,title,msg):
-        userInfo = QMessageBox.question(self,title,msg,
-                                        QMessageBox.Yes | QMessageBox.No)
+    def msgApp(self, title, msg):
+        userInfo = QMessageBox.question(
+            self, title, msg, QMessageBox.Yes | QMessageBox.No
+        )
         if userInfo == QMessageBox.Yes:
             return "Y"
         if userInfo == QMessageBox.No:
             return "N"
         self.close()
 
     def aboutHelp(self):
-        QMessageBox.about(self, "About Simple Molecule Editor",
-                """A Simple Molecule Editor where you can edit molecules\nBased on RDKit! http://www.rdkit.org/ \nSome icons from http://icons8.com\n\nSource code: https://github.com/EBjerrum/rdeditor""")
+        QMessageBox.about(
+            self,
+            "About Simple Molecule Editor",
+            """A Simple Molecule Editor where you can edit molecules\nBased on RDKit! http://www.rdkit.org/ \nSome icons from http://icons8.com\n\nSource code: https://github.com/EBjerrum/rdeditor""",
+        )
 
     def setAction(self):
         sender = self.sender()
         self.editor.setAction(sender.objectName())
-        self.myStatusBar.showMessage("Action %s selected"%sender.objectName())
+        self.myStatusBar.showMessage("Action %s selected" % sender.objectName())
+
+    def setRingType(self, ringType):
+        if ringType == self.editor.ringtype:
+            self.editor.setRingType(None)
+            self.bezeneAction.setChecked(False)
+            self.ringSixAction.setChecked(False)
+        else:
+            self.editor.setRingType(ringType)
+            if ringType != "benzene":
+                self.bezeneAction.setChecked(False)
+            if ringType != "aliphatic6":
+                self.ringSixAction.setChecked(False)
 
     def setBondType(self):
         sender = self.sender()
         self.editor.setBondType(sender.objectName())
-        self.myStatusBar.showMessage("Bondtype %s selected"%sender.objectName())
+        self.editor.setRingType(None)
```

Comment:
No, this should not be done here. I'm thinking that maybe the ringtype property implementation is not the way, as it creates an extra level of complexity in the code. I'm thinking that maybe we should rather modify the bond-type, to allow fragments (i.e. a bond or fragment type). The combination of a bondtype that is a ring or other fragment + a click on a bond or an atom, will allow us to delegate the edit action correctly.

Reply:
Seems it should be doable, when I look through the code. Let's discuss Thursday